evalutils/evalutils.py

```diff
@@ -5,18 +5,153 @@
 from os import PathLike
 from pathlib import Path
 from typing import Tuple, Dict, Set, Callable, List, Union
+
+try:
+    from typing import OrderedDict
+except ImportError:
+    from typing import MutableMapping
+
+    OrderedDict = MutableMapping
 from warnings import warn
 
 from pandas import DataFrame, merge, Series, concat
 
-from .exceptions import FileLoaderError, ValidationError, ConfigurationError
+from .exceptions import (
+    FileLoaderError,
+    ValidationError,
+    ConfigurationError,
+    FileLoaderIncludePatternError,
+)
 from .io import first_int_in_filename_key, FileLoader, CSVLoader
 from .scorers import score_detection
 from .validators import DataFrameValidator
 
 logger = logging.getLogger(__name__)
 
 
+class BaseProcess(ABC):
+    def __init__(
+        self,
+        *,
+        file_loaders: OrderedDict[str, FileLoader],
```

Comment:
That is because the first FileLoader will be used for indexing cases in `.process()`, unless it is explicitly specified. Alternatively, and maybe cleaner, we could use a normal `dict` for the file_loaders and add a separate `index_tag` which specifies the main file_loader's key.

Reply:
I applied my suggested changes.